packages/dynamics-lib/src/entities/base.entity.js

```diff
@@ -263,66 +274,63 @@ export class BaseEntity {
-          if (v instanceof BaseEntity) {
-            acc[k] = v.id ? `/${v.constructor.definition.dynamicsCollection}(${v.id})` : `$${v.uniqueContentId}`
-          } else {
-            acc[k] = v
-          }
-          return acc
-        }, {})
-      )
+    return Object.entries(this.constructor.definition.mappings).reduce(
+      (acc, [property, { field }]) => {
+        const serialized = this._toSerialized(property)
+        if (serialized !== undefined) {
+          acc[field] = serialized
+        }
+        return acc
+      },
+      Object.entries(this._bindings).reduce((acc, [k, v]) => {
+        if (v instanceof BaseEntity) {
+          acc[k] = v.id ? `/${v.constructor.definition.dynamicsCollection}(${v.id})` : `$${v.uniqueContentId}`
+        } else {
+          acc[k] = v
+        }
+        return acc
+      }, {})
+    )
   }
 
   /**
    * Create a new entity using the response from a query to the Dynamics ODATA Web API
    *
    * @param {string} etag the etag of the entity
-   * @param {Object} fields the fields of the entity
+   * @param {Object} entityData the data returned from Dynamics
    * @param {Object} optionSetData the global option set data used to resolve option set fields
    * @returns {BaseEntity} an instance of a BaseEntity subclass
    */
-  static fromResponse ({ '@odata.etag': etag, ...fields }, optionSetData) {
+  static fromResponse ({ '@odata.etag': etag, ...entityData }, optionSetData) {
     const instance = new this.prototype.constructor()
     instance._etag = etag
-    instance._localState = Object.entries(this.definition.mappings)
-      .filter(([property, { field }]) => fields[field] !== undefined)
-      .reduce((acc, [property, { field, type, ref }]) => {
-        let value = fields[field]
-        if (value !== null) {
-          if (type === 'integer' || type === 'decimal') {
-            value = Number(value)
-          } else if (type === 'optionset') {
-            const optionSetEntries = optionSetData[ref]
-            if (optionSetEntries) {
-              value = optionSetEntries.options[value]
-            } else {
-              throw new Error(`Unable to find optionset entries for ${ref}`)
-            }
+    instance._localState = Object.entries(this.definition.mappings).reduce((acc, [property, { field, type, ref }]) => {
+      let value = entityData[field]
+      if (value !== undefined && value !== null) {
+        if (type === 'integer' || type === 'decimal') {
+          value = Number(value)
+        } else if (type === 'optionset') {
+          const optionSetEntries = optionSetData[ref]
+          if (optionSetEntries) {
+            value = optionSetEntries.options[value]
+          } else {
+            throw new Error(`Unable to find optionset entries for ${ref}`)
           }
         }
+      }
+      if (value !== undefined) {
         acc[property] = value
-        return acc
-      }, {})
+      }
+      return acc
+    }, {})
```

Comment:
Again, a bit tricky to work out what's going on

Reply:
Yes, agreed.  Some additional comments might help to describe what the method is doing